File: backend/src/main/java/com/nottingham/mynottingham/backend/dto/ForumDto.java
package com.nottingham.mynottingham.backend.dto;

import com.nottingham.mynottingham.backend.entity.ForumPost;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * DTOs for Forum feature
 */
public class ForumDto {

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ForumPostDto {
        private Long id;
        private Long authorId;
        private String authorName;
        private String authorAvatar;
        private String title;
        private String content;
        private String category;
        private String imageUrl;
        private Integer likes;
        private Integer comments;
        private Integer views;
        private Boolean isPinned;
        private Boolean isLocked;
        private Boolean isLikedByCurrentUser;
        private List<String> tags;
        private LocalDateTime createdAt;
        private LocalDateTime updatedAt;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ForumCommentDto {
        private Long id;
        private Long postId;
        private Long authorId;
        private String authorName;
        private String authorAvatar;
        private String content;
        private Integer likes;
        private Boolean isLikedByCurrentUser;
        private LocalDateTime createdAt;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class CreateForumPostRequest {
        private String title;
        private String content;
        private ForumPost.ForumCategory category;
        private List<String> tags;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class UpdateForumPostRequest {
        private String title;
        private String content;
        private ForumPost.ForumCategory category;
        private List<String> tags;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class CreateCommentRequest {
        private String content;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class PagedForumResponse {
        private List<ForumPostDto> posts;
        private Integer currentPage;
        private Integer totalPages;
        private Long totalElements;
        private Boolean hasNext;
        private Boolean hasPrevious;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ForumPostDetailDto {
        private ForumPostDto post;
        private List<ForumCommentDto> comments;
    }
}

File: app/src/main/java/com/nottingham/mynottingham/data/remote/dto/ForumDto.kt
package com.nottingham.mynottingham.data.remote.dto

import java.time.LocalDateTime

/**
 * Data Transfer Objects for Forum feature
 * Maps to backend ForumDto.java
 */

// Response DTOs
data class ForumPostDto(
    val id: Long,
    val authorId: Long,
    val authorName: String,
    val authorAvatar: String?,
    val title: String,
    val content: String,
    val category: String,
    val imageUrl: String?,
    val likes: Int,
    val comments: Int,
    val views: Int,
    val isPinned: Boolean,
    val isLocked: Boolean,
    val isLikedByCurrentUser: Boolean,
    val tags: List<String>?,
    val createdAt: String,  // ISO 8601 format from backend
    val updatedAt: String
)

data class ForumCommentDto(
    val id: Long,
    val postId: Long,
    val authorId: Long,
    val authorName: String,
    val authorAvatar: String?,
    val content: String,
    val likes: Int,
    val isLikedByCurrentUser: Boolean,
    val createdAt: String  // ISO 8601 format from backend
)

data class PagedForumResponse(
    val posts: List<ForumPostDto>,
    val currentPage: Int,
    val totalPages: Int,
    val totalElements: Long,
    val hasNext: Boolean,
    val hasPrevious: Boolean
)

data class ForumPostDetailDto(
    val post: ForumPostDto,
    val comments: List<ForumCommentDto>
)

// Request DTOs
data class CreateForumPostRequest(
    val title: String,
    val content: String,
    val category: String,
    val tags: List<String>?
)

data class UpdateForumPostRequest(
    val title: String,
    val content: String,
    val category: String,
    val tags: List<String>?
)

data class CreateCommentRequest(
    val content: String
)

// API Response wrappers
data class ForumPostResponse(
    val success: Boolean,
    val message: String,
    val data: ForumPostDto?
)

data class ForumPostDetailResponse(
    val success: Boolean,
    val message: String,
    val data: ForumPostDetailDto?
)

data class PagedForumPostsResponse(
    val success: Boolean,
    val message: String,
    val data: PagedForumResponse?
)

data class ForumCommentResponse(
    val success: Boolean,
    val message: String,
    val data: ForumCommentDto?
)
